src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Helpers/RemoveUnnecessaryImports/CSharpUnnecessaryImportsProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.RemoveUnnecessaryImports
+{
+    internal sealed class CSharpUnnecessaryImportsProvider
```

Comment:
This provider contains existing code split out from [AbstractCSharpRemoveUnnecessaryImportsService](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/RemoveUnnecessaryImports/AbstractCSharpRemoveUnnecessaryImportsService.cs,47)